source/NVDAObjects/IAccessible/sysListView32.py

```diff
@@ -394,7 +394,11 @@ def _get_name(self):
 				textList.append("%s: %s" % (header, content))
 			else:
 				textList.append(content)
-		return "; ".join(textList)
+		name = "; ".join(textList)
+		#Some list view items in Vista and later can contain annoying left-to-right and right-to-left indicator characters which really should not be there.
+		name=name.replace(u'\u200E','')

```

Comment:
> Wouldn't it be better to just return the modified name in one line like return name.replace(blah, "").replace(blah,"")?

 

I would say yes, but this code was copied from another function in that file that dealt with the same issue, so my changes were meant to keep it consistent.

 

From: josephsl [mailto:notifications@github.com] 
Sent: Sunday, April 3, 2016 10:16
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Davy Kager <mail@davykager.nl>
Subject: Re: [nvaccess/nvda] Improve reading of the clock in the taskbar (#5751)

 

In source/NVDAObjects/IAccessible/sysListView32.py <https://github.com/nvaccess/nvda/pull/5751#discussion_r58306171> :

> @@ -394,7 +394,11 @@ def _get_name(self):
>                              textList.append("%s: %s" % (header, content))
>                      else:
>                              textList.append(content)
> -            return "; ".join(textList)
> +            name = "; ".join(textList)
> +            #Some list view items in Vista and later can contain annoying left-to-right and right-to-left indicator characters which really should not be there.
> +            name=name.replace(u'\u200E','')

Wouldn't it be better to just return the modified name in one line like return name.replace(blah, "").replace(blah,"")?

—
You are receiving this because you authored the thread.
Reply to this email directly or  <https://github.com/nvaccess/nvda/pull/5751/files/19b600d36da6a6f2557663dc605a251fad2086c6#r58306171> view it on GitHub  <https://github.com/notifications/beacon/AC9y8Vco0_lzDo1Y2pmig7vaakSKjxwmks5pz3dIgaJpZM4HZsGx.gif>